test/generator-test.js

```diff
@@ -166,21 +168,20 @@ suite("Rules Generator Tests", function() {
       { f: "",
         x: "this.foo || this.bar",
         expect: "newData.child('foo').val() == true || newData.child('bar').val() == true"},
-      // Don't support snapshot functions beyond parent.
+      // TODO: Don't support snapshot functions beyond parent.
       // TODO: Should warn user not to use Firebase builtins!
-      { f: "", x: "this.isString == 'a'", expect: "newData.child('isString').val() == 'a'" },
+      // { f: "", x: "this.isString()", expect: "newData.child('isString').val() == true" },
```

Comment:
That's what the TODO comment is about ... because I end up reprocessing expressions, I have the unfortunate problem that I sometimes expand them when they are user-provided only (in which case I was to restrict the use of deprecated Firebase functions), or in processing post-processed expressions, which DO include use of these Firebase functions.

The current code does not distinguish these cases - so I am left with a TODO and commented out test.